src/ensembl/src/shared/hooks/useOutsideClick.tsx

```diff
@@ -17,17 +17,47 @@
 import { useEffect } from 'react';
 
 export default function useOutsideClick<T extends HTMLElement>(
-  ref: React.RefObject<T>,
+  refOrRefs: React.RefObject<T> | React.RefObject<T>[],
   callback: () => void
 ) {
+  const refs = Array.isArray(refOrRefs) ? refOrRefs : [refOrRefs];
+
+  let clickedInside = false;
+
   const handleClickOutside = (event: Event) => {
-    if (ref.current && !ref.current.contains(event.target as HTMLElement)) {
+    let clickedOutside = true;
+
+    if (clickedInside) {
+      // Reset the clickedInside flag to false
+      clickedInside = false;
+      return;
+    }
+
+    refs.forEach((ref) => {
+      if (ref.current && ref.current.contains(event.target as HTMLElement)) {
+        clickedOutside = false;
+      }
+    });
```

Comment:
You can probably do this even without having to introduce the `clickedOutside` variable.

```
for (const ref of refs) {
  if (ref.current && ref.current.contains(event.target as HTMLElement)) {
    callback();
    break;
  }
}
```

Reply:
That's done